pkgs/tools/admin/incus/default.nix

```
@@ -0,0 +1,86 @@
+{
+  lib,
+  hwdata,
+  pkg-config,
+  lxc,
+  buildGoModule,
+  fetchurl,
+  acl,
+  libcap,
+  cowsql,
+  raft-canonical,
+  sqlite,
+  udev,
+  installShellFiles,
+  nixosTests,
+  gitUpdater,
+  callPackage,
+}:
+buildGoModule rec {
+  pname = "incus";
+  version = "0.1";
+
+  src = fetchurl {
+    url = "https://github.com/lxc/incus/releases/download/incus-${version}/incus-${version}.tar.gz";
+    hash = "sha256-dxtvQ442n3Keqqbf2TjxTcyN4J28naI8fIDG0U8CZTs=";
+  };
+
+  vendorHash = null;
+
+  postPatch = ''
+    substituteInPlace internal/usbid/load.go \
+      --replace "/usr/share/misc/usb.ids" "${hwdata}/share/hwdata/usb.ids"
+  '';
+
+  excludedPackages = ["test" "incus/internal/server/db/generate" "incus-agent" "incus-migrate" "cmd/lxd-to-incus"];
+
+  nativeBuildInputs = [installShellFiles pkg-config];
+  buildInputs = [
+    lxc
+    acl
+    libcap
+    cowsql.dev # TODO
+    raft-canonical.dev
+    sqlite
+    udev.dev
+  ];
+
+  ldflags = ["-s" "-w"];
+  tags = ["libsqlite3"];
+
+  preBuild = ''
+    # required for go-dqlite. See: https://github.com/canonical/lxd/pull/8939
+    export CGO_LDFLAGS_ALLOW="(-Wl,-wrap,pthread_create)|(-Wl,-z,now)"
+  '';
+
+  # build static binaries: https://github.com/canonical/lxd/blob/6fd175c45e65cd475d198db69d6528e489733e19/Makefile#L43-L51
+  postBuild = ''
+    make incus-agent incus-migrate
+  '';
+
+  preCheck = let
+    skippedTests = [
+      "TestValidateConfig"
+      "TestConvertNetworkConfig"
+      "TestConvertStorageConfig"
+      "TestSnapshotCommon"
+      "TestContainerTestSuite"
+    ];
+  in ''
+    # Disable tests requiring local operations
```

comments inside of smart strings like this is an antipattern, because changing them requires a rebuild, if you move it outside of this, that would no longer be the case.
```suggestion
  # Disable tests requiring local operations
  in ''
```